File: Modulo4-ObjectiveC-Lorena/Exercicio2.m
//
//  main.m
//  Exercicio2-C
//
//  Created by Lorena Rodrigues Bruno on 10/09/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

#import <Foundation/Foundation.h>

@interface Motor : NSObject {
    int potencia;
}
@property(nonatomic, readwrite) int potencia;
@end

@implementation Motor
@synthesize potencia;
-(id) initWithPotencia : (int)potencia {
    self = [super init];
    if (potencia < 0) {
        self.potencia = 0;
    } else if (potencia > 100){
        self.potencia = 100;
    } else {
        self.potencia = potencia;
    }
    return self;
}
@end

int main(int argc, const char * argv[]) {
    @autoreleasepool {
        Motor *motor1 = [[Motor alloc]initWithPotencia:-50];
        printf("Potencia do motor 1: %i\n", motor1.potencia);
        
        Motor *motor2 = [[Motor alloc]initWithPotencia:60];
        printf("Potencia do motor 2: %i\n", motor2.potencia);
        
        Motor *motor3 = [[Motor alloc]initWithPotencia:130];
        printf("Potencia do motor 3: %i\n", motor3.potencia);
        
    }
    return 0;
}

File: Modulo4-Lorena/Exercicio2.swift
//
//  main.swift
//  Exercicio1
//
//  Created by Lorena Rodrigues Bruno on 28/08/2018.
//  Copyright © 2018 Lorena Rodrigues Bruno. All rights reserved.
//

import Foundation

class CarroCorrida {
    var numeroIdentificao = Int()
    var velocidadeAtual = 0.0
    var velocidadeMaxima = 100.0
    
    init(velocidadeMaxima : Double) {
        
        self.velocidadeMaxima = velocidadeMaxima
    }
    
}

//testes:
//var carro1 = CarroCorrida(velocidadeMaxima: 160)
//print("Velocidade Máxima: \(carro1.velocidadeMaxima)")


